triton-vm/src/table/processor_table.rs

```diff
@@ -1670,6 +1671,52 @@ impl ExtProcessorTable {
         .concat()
     }
 
+    fn instruction_recurse_or_return(
+        circuit_builder: &ConstraintCircuitBuilder<DualRowIndicator>,
+    ) -> Vec<ConstraintCircuitMonad<DualRowIndicator>> {
+        let one = || circuit_builder.b_constant(1);
+        let curr_row = |col: ProcessorBaseTableColumn| {
+            circuit_builder.input(CurrentBaseRow(col.master_base_table_index()))
+        };
+        let next_row = |col: ProcessorBaseTableColumn| {
+            circuit_builder.input(NextBaseRow(col.master_base_table_index()))
+        };
+
+        // Zero if the ST5 equals ST6. One if they are not equal.
+        let st5_eq_st6 = || curr_row(HV0) * (curr_row(ST6) - curr_row(ST5));
+        let st5_neq_st6 = || one() - st5_eq_st6();
```

Comment:
What's the motivation for defining these as closures rather than constructing nodes and `clone`ing them?

Reply:
Now, mostly alignment for the code below: I like the looks of

```rust
let maybe_return = vec![
    st5_neq_st6() * curr_row(HV0),
    st5_neq_st6() * (curr_row(ST6) - curr_row(ST5)),
    st5_neq_st6() * (next_row(IP) - curr_row(JSO)),
    st5_neq_st6() * (next_row(JSP) - curr_row(JSP) + one()),
];
```

better than

```rust
let maybe_return = vec![
    st5_neq_st6.clone() * curr_row(HV0),
    st5_neq_st6.clone() * (curr_row(ST6) - curr_row(ST5)),
    st5_neq_st6.clone() * (next_row(IP) - curr_row(JSO)),
    st5_neq_st6 * (next_row(JSP) - curr_row(JSP) + one()),
];
```

because in my opinion, it provides greates visual clarity that the first part of the constraint is indeed identical for all four constraints.

Historically, it was also useful to pass the stack element in question as an argument to the closure.